src/core/src/runtime/allocator.cpp

```diff
@@ -11,19 +11,66 @@
 
 namespace ov {
 
-Allocator::Allocator() : _impl{std::make_shared<BlobAllocator>()} {}
+struct DefaultAllocator {
+    void* allocate(const size_t bytes, const size_t alignment) {
+        if (alignment == alignof(max_align_t)) {
+            return ::operator new(bytes);
+        } else {
+            OPENVINO_ASSERT((alignment % 2) == 0, "Alignment is not power of 2: ", alignment);
```

Comment:
This is even number check not power of 2.
The power of two for unsigned or not negative number could be done like
```cpp
template<class T>
constexpr bool is_power_of_two(T n){
    return n && !static_cast<bool>(n & (n - static_cast<T>(1)));
}
```